crates/term/src/backends/process.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (C) 2024 Lily Lyons
+//
+// This file is part of Luminol.
+//
+// Luminol is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Luminol is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Luminol.  If not, see <http://www.gnu.org/licenses/>.
+//
+//     Additional permission under GNU GPL version 3 section 7
+//
+// If you modify this Program, or any covered work, by linking or combining
+// it with Steamworks API by Valve Corporation, containing parts covered by
+// terms of the Steamworks API by Valve Corporation, the licensors of this
+// Program grant you additional permission to convey the resulting work.
+
+use std::sync::{
+    mpsc::{Receiver, Sender},
+    Arc,
+};
+
+use alacritty_terminal::{
+    event::{Event, Notify, WindowSize},
+    event_loop::{Msg, Notifier},
+    grid::Dimensions,
+    sync::FairMutex,
+    term::{test::TermSize, Term},
+};
+
+use super::Backend;
+
+pub struct Process {
+    term: Arc<FairMutex<Term<ForwardEventListener>>>,
+    notifier: Notifier,
+    event_reciever: Receiver<Event>,
+}
+
+#[derive(Clone)]
+pub struct ForwardEventListener(Sender<Event>, egui::Context);
+
+impl alacritty_terminal::event::EventListener for ForwardEventListener {
+    fn send_event(&self, event: Event) {
+        if matches!(event, Event::Wakeup) {
+            self.1.request_repaint();
+        }
+        let _ = self.0.send(event);
+    }
```

Comment:
Actually, the repaint should probably happen after the sending to avoid race conditions